logstreams/src/test/java/io/zeebe/logstreams/log/LogBlockIndexTest.java

```diff
@@ -15,85 +15,82 @@
 
   @Before
-  public void setup() {
-    blockIndex = createNewBlockIndex(CAPACITY);
-  }
+  public void setup() throws Exception {
+    runtimeDirectory.create();
+    snapshotDirectory.create();
 
-  protected LogBlockIndex createNewBlockIndex(int capacity) {
-    return new LogBlockIndex(capacity, c -> new UnsafeBuffer(ByteBuffer.allocate(c)));
+    startBlockIndexDb();
   }
 
-  @Test
-  public void shouldAddBlock() {
-    final int capacity = blockIndex.capacity();
-
-    // when
+  private void startBlockIndexDb() throws Exception {
+    final ZeebeDbFactory<LogBlockColumnFamilies> dbFactory =
+        ZeebeRocksDbFactory.newFactory(LogBlockColumnFamilies.class);
 
-    for (int i = 0; i < capacity; i++) {
-      final int pos = i + 1;
-      final int addr = pos * 10;
-      final int expectedSize = i + 1;
+    final FsSnapshotStorageConfiguration config = new FsSnapshotStorageConfiguration();
+    config.setRootPath(snapshotDirectory.getRoot().getAbsolutePath());
 
-      assertThat(blockIndex.addBlock(pos, addr)).isEqualTo(expectedSize);
-      assertThat(blockIndex.size()).isEqualTo(expectedSize);
-    }
+    final LogStreamBuilder builder =
+        new LogStreamBuilder(0)
+            .logName(NAME)
+            .blockIndexDirectory(runtimeDirectory.getRoot())
+            .snapshotDirectory(snapshotDirectory.getRoot())
+            .snapshotConfig(config);
 
-    // then
+    blockIndex = new LogBlockIndex(dbFactory, builder);
+    blockIndex.recoverFromSnapshot(NAME);
+    blockIndex.openDb();
+  }
 
-    for (int i = 0; i < capacity; i++) {
-      final int virtPos = i + 1;
-      final int physPos = virtPos * 10;
+  @After
+  public void tearDown() {
+    runtimeDirectory.delete();
+    snapshotDirectory.delete();
 
-      assertThat(blockIndex.getLogPosition(i)).isEqualTo(virtPos);
-      assertThat(blockIndex.getAddress(i)).isEqualTo(physPos);
+    try {
+      blockIndex.closeDb();
+    } catch (final Exception e) {
+      e.printStackTrace();
     }
   }
 
   @Test
-  public void shouldExpandAndAddBlockIfCapacityReached() {
-    // given
-    final int capacity = blockIndex.capacity();
-    int elements = 0;
-    final Function<Integer, Long> addrFunc = x -> x * 10L;
-
-    for (; elements <= capacity; elements++) {
-      blockIndex.addBlock(elements, addrFunc.apply(elements));
-    }
+  public void shouldAddBlocks() {
+    final int numBlocks = 10;
 
     // when
-    for (; elements < capacity + 5; elements++) {
-      blockIndex.addBlock(elements, addrFunc.apply(elements));
-    }
+    final long lastPosition = addBlocks(numBlocks);
 
     // then
-    assertThat(blockIndex.size()).isEqualTo(elements);
-    assertThat(blockIndex.capacity()).isEqualTo(Math.round(capacity * GROW_FACTOR));
-
-    for (int i = 0; i < elements; i++) {
-      assertThat(blockIndex.getAddress(i)).isEqualTo(addrFunc.apply(i));
-    }
+    lookupAndAssert(numBlocks);
+    assertThat(lastPosition).isEqualTo(lastPosition);
```

Comment:
this will always be equal right? :sweat_smile:

Reply:
please assert that blockIndex.isEmpty == false

Reply:
:man_facepalming: indeed, the first should be getLastPosition()